sklearn/manifold/_locally_linear.py

```diff
@@ -367,7 +367,10 @@ def locally_linear_embedding(
         Yi = np.empty((n_neighbors, 1 + n_components + dp), dtype=np.float64)
         Yi[:, 0] = 1
 
-        M = np.zeros((N, N), dtype=np.float64)
+        if M_sparse:
```

Comment:
before the different `if method == ...`, we could create the type of constructor:

```python
    M_sparse = eigen_solver != "dense"
    M_container_constructor = lil_maxtrix if M_sparse else np.zeros
```

Then in the different branches we will have:

```python
    M = M_container_constructor ((N, N), dtype=np.float64)
```